Pecl install on mac (with homebrew) cannot use system lz4

### Expected behaviour

When I run this command:

```shell
pecl install --configureoptions 'enable-redis-igbinary="no" enable-redis-lzf="no" enable-redis-zstd="no" enable-redis-msgpack="no" enable-redis-lz4="yes" with-liblz4="/opt/homebrew/Cellar/lz4/1.9.4"' redis
```

I expect that either I'll get a configure error when it can't find the library (`configure: error: Please reinstall the liblz4 distribution`), or the package will build and install successfully.

### Actual behaviour

The lz4 library is detected successfully, and pecl proceeds to the build step, but then this happens:

```
...
/private/tmp/pear/temp/redis/redis.c:48:10: fatal error: 'lz4.h' file not found
#include <lz4.h>
         ^~~~~~~
1 error generated.
make: *** [redis.lo] Error 1
ERROR: `make' failed
```

### I'm seeing this behaviour on
- OS: Macos Sonoma 14.3.1
- Redis: as published in pecl
- PHP: 8.2
- phpredis: as published in pecl

### Steps to reproduce, backtrace or example script

### I've checked
- [x] There is no similar issue from other users
- [x] ~~Issue isn't fixed in `develop` branch~~ can't pecl install from develop

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,6 +1,72 @@
 on: [push, pull_request]
 
 jobs:
+  configured-deps:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    strategy:
+      fail-fast: true
+      matrix:
+        php: ['8.3']
+    steps:
+      - name: Checkout PhpRedis
+        uses: actions/checkout@v4
+
+      - name: Install liblzf
+        run: |
+          git clone --depth=1 https://github.com/nemequ/liblzf.git
+          cd liblzf
+          autoreconf -vi
+          CFLAGS=-fPIC ./configure --prefix="$GITHUB_WORKSPACE/liblzf"
+          make install
+
+      - name: Install liblz4
+        run: |
+          git clone -b v1.9.4 --depth=1 https://github.com/lz4/lz4
+          cd lz4/lib
+          PREFIX="$GITHUB_WORKSPACE/liblz4" make install
+
+      - name: Install libzstd
+        run: |
+          git clone -b v1.5.5 --depth=1 https://github.com/facebook/zstd
+          cd zstd
+          PREFIX="$GITHUB_WORKSPACE/libzstd" make install
+
+      - name: Install PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: json, igbinary, msgpack, :redis
+          coverage: none
+          tools: none
+
+      - name: Configure and build PhpRedis with distinct dep paths
+        run: |
+          phpize
+          ./configure \
+            --enable-redis-lz4 \
+            --with-liblz4="$GITHUB_WORKSPACE/liblz4" \
+            --enable-redis-lzf \
+            --with-liblzf="$GITHUB_WORKSPACE/liblzf" \
+            --enable-redis-zstd \
+            --with-libzstd="$GITHUB_WORKSPACE/libzstd"
+          sudo make -j"$(nproc)"
+
+      - name: Make sure we're linking against specific liblz4
+        run: |
+          grep "INCLUDES.*$GITHUB_WORKSPACE/liblz4" Makefile
+          grep "REDIS_SHARED_LIBADD.*-L$GITHUB_WORKSPACE/liblz4" Makefile
+
+      - name: Make sure we're linking against specific liblzf
+        run: |
+          grep "INCLUDES.*$GITHUB_WORKSPACE/liblzf" Makefile
+          grep "REDIS_SHARED_LIBADD.*-L$GITHUB_WORKSPACE/liblzf" Makefile
+
+      - name: Make sure we're linking against specific libzstd
+        run: |
+          grep "INCLUDES.*$GITHUB_WORKSPACE/libzstd" Makefile
+          grep "REDIS_SHARED_LIBADD.*-L$GITHUB_WORKSPACE/libzstd" Makefile
+
   ubuntu:
     runs-on: ubuntu-latest
     continue-on-error: false

--- a/config.m4
+++ b/config.m4
@@ -221,6 +221,7 @@ if test "$PHP_REDIS" != "no"; then
       PHP_CHECK_LIBRARY(lzf, lzf_compress,
       [
         PHP_ADD_LIBRARY_WITH_PATH(lzf, $LIBLZF_DIR/$PHP_LIBDIR, REDIS_SHARED_LIBADD)
+        PHP_ADD_INCLUDE($LIBLZF_DIR/include)
       ], [
         AC_MSG_ERROR([could not find usable liblzf])
       ], [
@@ -263,12 +264,12 @@ if test "$PHP_REDIS" != "no"; then
       PHP_CHECK_LIBRARY(lz4, LZ4_compress,
       [
         PHP_ADD_LIBRARY_WITH_PATH(lz4, $LIBLZ4_DIR/$PHP_LIBDIR, REDIS_SHARED_LIBADD)
+        PHP_ADD_INCLUDE($LIBLZ4_DIR/include)
       ], [
         AC_MSG_ERROR([could not find usable liblz4])
       ], [
         -L$LIBLZ4_DIR/$PHP_LIBDIR
       ])
-      PHP_SUBST(REDIS_SHARED_LIBADD)
     else
       AC_MSG_ERROR([only system liblz4 is supported])
     fi
@@ -307,6 +308,7 @@ if test "$PHP_REDIS" != "no"; then
       PHP_CHECK_LIBRARY(zstd, ZSTD_getFrameContentSize,
       [
         PHP_ADD_LIBRARY_WITH_PATH(zstd, $LIBZSTD_DIR/$PHP_LIBDIR, REDIS_SHARED_LIBADD)
+        PHP_ADD_INCLUDE($LIBZSTD_DIR/include)
       ], [
         AC_MSG_ERROR([could not find usable libzstd, version 1.3.0 required])
       ], [

